### PR TITLE
ci: set up basic workflows

### LINKED ISSUE #2: Initialize the `ratify-go` repository
The `ratify-go` repository is the go library for the Ratify Core.

To make the repository contributor-friendly, the following issues should be addressed:
- [ ] Set up a `MAINTAINERS` file
- [ ] Set up a `CODEOWNERS` file to reflect `MAINTAINERS`
- [x] Set up code coverage check
- [x] Set up dependabot
- [x] Set up license check
- [x] Set up CodeQL analysis
- [x] Set up PR build check
- [ ] Set up `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by shizhMSFT**
Partially resolves #2

Changes:
- Revise license and add license checker
- Add `go.mod`
- Add go package doc
- Add dependabot
- Add PR test gate
- Add codecov
- Add CodeQL
- Add `.gitignore`

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2021 Ratify Authors.
```
Comment (shizhMSFT) [inline LICENSE:189]: This is copied from https://github.com/ratify-project/ratify/blob/dev/LICENSE#L189

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+on:
+  push:
+    branches: 
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+
```
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply .github/workflows/build.yml:25]: specified more strict permissions

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      checks: write
+    strategy:
+      matrix:
+        go-version: ['1.22', '1.23']
+      fail-fast: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Go ${{ matrix.go-version }} environment
+        uses: actions/setup-go@v5
```
Comment (binbin-li) [inline .github/workflows/build.yml:41]: in ratify repo, we usually pin to a specific version and add tag as comment.
Like: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0

File: .github/workflows/codeql.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  push:
+    branches:
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+  schedule:
+    - cron: '19 7 * * 1'
```
Comment (binbin-li) [inline .github/workflows/codeql.yml:26]: any reason for this time setup?

Comment (shizhMSFT) [reply .github/workflows/codeql.yml:26]: This time is random

File: .github/workflows/build.yml (hunk 3 above)
Comment (shizhMSFT) [reply .github/workflows/build.yml:41]: Yepp, this will be the next step.

**Review by binbin-li — APPROVED** [review]

**Pull request merged**